File: limeutils/src/main/java/com/xlib/limeutils/json/JsonLoader.java
package com.xlib.limeutils.json;

import android.content.Context;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;
import java.io.InputStream;
import java.nio.charset.StandardCharsets;

public class JsonLoader {

    private static final String TAG = "JsonLoader";

    private Context context;
    //private String fileName;

//    private static final int DIR_ASSET = 1;
//    private static final int DIR_RAW = 2;
//    private static final int DIR_EXTERNAL_STORAGE = 3;

    private JsonLoader(Context context) {
        this.context = context;
    }

    public static JsonLoader with(Context applicationContext) {
        return new JsonLoader(applicationContext);
    }

//    public JsonLoader fileName(String fileName) {
//        this.fileName = fileName;
//        return this;
//    }

    public String getAsString(String fileName) {
        String json = null;
        try {
            InputStream inputStream = getInputStream(fileName);
            int size = inputStream.available();
            byte[] buffer = new byte[size];
            //noinspection ResultOfMethodCallIgnored
            inputStream.read(buffer);
            inputStream.close();
            json = new String(buffer, StandardCharsets.UTF_8);
        } catch (IOException e) {
            e.printStackTrace();
        }
        return json;
    }

    public String getAsString(int res) {
        String json = null;
        try {
            InputStream inputStream = getInputStream(res);
            int size = inputStream.available();
            byte[] buffer = new byte[size];
            //noinspection ResultOfMethodCallIgnored
            inputStream.read(buffer);
            inputStream.close();
            json = new String(buffer, StandardCharsets.UTF_8);
        } catch (IOException e) {
            e.printStackTrace();
        }
        return json;
    }


    public JSONObject getAsJSONObject(String fileName) {

        String json = getAsString(fileName);
        JSONObject jsonObject = null;
        try {
            jsonObject = new JSONObject(json);
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return jsonObject;
    }

    public JSONArray getAsJSONArray(String fileName) {
        String json = getAsString(fileName);
        JSONArray jsonArray = null;
        try {
            jsonArray = new JSONArray(json);
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return jsonArray;
    }

    private InputStream getInputStream(String fileName) throws IOException {

        InputStream inputStream = context.getAssets().open(fileName);
        return inputStream;
    }

    private InputStream getInputStream(int res) throws IOException {

        InputStream inputStream = context.getResources().openRawResource(res);
        return inputStream;
    }



}

File: limeutils/src/main/java/com/xlib/limeutils/json/JsonLoader.kt
package com.xlib.limeutils.json

import android.content.Context

import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

import java.io.IOException
import java.io.InputStream
import java.nio.charset.StandardCharsets

class JsonLoader private constructor(private val context: Context) {

    companion object {

        private val TAG = "JsonLoader"

        fun with(applicationContext: Context): JsonLoader {
            return JsonLoader(applicationContext)
        }
    }


    fun getAsString(fileName: String): String? {
        var json: String? = null
        try {
            val inputStream = getInputStream(fileName)
            val size = inputStream.available()
            val buffer = ByteArray(size)

            inputStream.read(buffer)
            inputStream.close()
            json = String(buffer, StandardCharsets.UTF_8)
        } catch (e: IOException) {
            e.printStackTrace()
        }

        return json
    }

    fun getAsString(res: Int): String? {
        var json: String? = null
        try {
            val inputStream = getInputStream(res)
            val size = inputStream.available()
            val buffer = ByteArray(size)

            inputStream.read(buffer)
            inputStream.close()
            json = String(buffer, StandardCharsets.UTF_8)
        } catch (e: IOException) {
            e.printStackTrace()
        }

        return json
    }


    fun getAsJSONObject(fileName: String): JSONObject? {

        val json = getAsString(fileName)
        var jsonObject: JSONObject? = null
        try {
            jsonObject = JSONObject(json!!)
        } catch (e: JSONException) {
            e.printStackTrace()
        }

        return jsonObject
    }

    fun getAsJSONArray(fileName: String): JSONArray? {
        val json = getAsString(fileName)
        var jsonArray: JSONArray? = null
        try {
            jsonArray = JSONArray(json)
        } catch (e: JSONException) {
            e.printStackTrace()
        }

        return jsonArray
    }

    @Throws(IOException::class)
    private fun getInputStream(fileName: String): InputStream {
        return context.assets.open(fileName)
    }

    @Throws(IOException::class)
    private fun getInputStream(res: Int): InputStream {
        return context.resources.openRawResource(res)
    }




}
